docs/arcadia-users-group/20220912-intro-to-shell2/lesson.md

```diff
@@ -0,0 +1,301 @@
+```
+wc -l
+```
+
+The options below allow you to select which counts are printed.
+
+* `-l`, `--lines` - print the number of lines.
+* `-w`, `--words` - print the number of words.
+* `-m`, `--chars` - print the number of characters.
+* `-c`, `--bytes` - print the number of bytes.
+* `-L`, `--max-line-length` - print the length of the longest line.
+
+## Copying, moving, and renaming files
+
+The commands **`cp`** and **`mv`** (**c**o**p**y and **m**o**v**e) have the same basic structure. 
+They both require two positional arguments – the first is the file you want to act on, and the second is where you want it to go (which can include the name you want to give it). 
+
+
+To see how this works, let's make a copy of "example.txt":
+
+```
+ls
+cp mkdocs.yml mkdocs_copy.yml
+ls
+```
+
+By giving the second argument a name and nothing else (meaning no path in front of the name), we are implicitly saying we want it copied to where we currently are. 
+
+To make a copy and put it somewhere else, like in our subdirectory `docs`, we could change the second positional argument using a **relative path** ("relative" because it starts from where we currently are):
+
+```
+ls docs
+cp mkdocs.yml docs/mkdocs_copy.yml
+ls docs
+```
+
+To copy a file to that subdirectory but keep the same name, we could type the whole name out, but we can also provide the directory but leave off the file name:
+
+```
+cp mkdocs.yml docs/
+ls docs/
+```
+
+If we wanted to copy something *from somewhere else to our current working directory* and keep the same name, we can use another special character, a period (**`.`**), which specifies the current working directory:
+
+```bash
+ls
+cp docs/index.md .
+ls
+```
+
+The **`mv`** command is used to move files. 
+Let's move the `README.md` file into the `docs` subdirectory:
+
+```
+ls
+ls docs/
+mv README.md docs/
+ls
+ls docs/
+```
+
+The **`mv`** command is also used to *rename* files. 
+This may seem strange at first, but remember that the path (address) of a file actually includes its name too (otherwise everything in the same directory would have the same path). 
+
+```
+ls
+mv mkdocs_copy.yml mkdocs_new.yml
+ls
+```
+
+To delete files there is the **`rm`** command (**r**e**m**ove). 
+This command requires at least one argument specifying the file we want to delete. 
+Typically, this command does not ask you to confirm your actions and files are permanently deleted instead of being moved to a `Trash` directory.
+
+```
+ls
+rm mkdocs_new.yml
+ls
+```
+
+## Creating and removing directories
+
+We can make a new directory with the command **`mkdir`** (for **m**a**k**e **dir**ectory): 
+
+```
+ls
+mkdir subset
+ls
+```
+
+And similarly, directories can be deleted with **`rmdir`** (for **r**e**m**ove **dir**ectory):
+
+```bash
+rmdir subset/
+ls
+```
+
+The command line is a little more forgiving when trying to delete a directory. 
+If the directory is not empty, **`rmdir`** will give you an error. 
+
+```bash
+rmdir docs/
```

Comment:
I actually never use this and almost always use `rm -rf` for removing a directory. Worth mentioning perhaps.